File: Add Code Here/LeetCode_Sols/Easy/145. Binary Tree Postorder Traversal/Solution.java
class Solution {
    List<Integer> order = new ArrayList<Integer>();
    public List<Integer> postorderTraversal(TreeNode root) {
        traverse(root);
        return order;
    }
    
    public void traverse(TreeNode root){
        if(root == null) return;
        traverse(root.left);
        traverse(root.right);
        order.add(root.val);
    }
}

File: Add Code Here/LeetCode_Sols/Easy/145. Binary Tree Postorder Traversal/Solution.kt
/**
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int = 0) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun postorderTraversal(root: TreeNode?): List<Int> {
        val result = mutableListOf<Int>()
        if (root == null) return result

        result.addAll(postorderTraversal(root.left))
        result.addAll(postorderTraversal(root.right))
        result.add(root.`val`)

        return result
    }
}